dev/storage_base.h

```diff
@@ -728,6 +741,30 @@ namespace sqlite_orm {
                 }
             }
 
+            // migration internal API
+            void start_migration(bool turn_off_fk) {
+                /*
+                 * this #define enables or disable FK checking OFF as part of the migration
+                 * undefined => no FK checking OFF EVER!!
+                 */
+#ifdef FK_TOGGLE_ENABLE
+                this->pragma.foreign_keys(turn_off_fk ? false : true);
+#endif
+                this->begin_exclusive_transaction();
+            }
+            void commit_migration() {
+                auto foreign_keys_guard = finally([&pragma = this->pragma]() {
+                    pragma.foreign_keys(true);
+                });
+                this->commit();
+            }
+            void abort_migration() {
+                auto foreign_keys_guard = finally([&pragma = this->pragma]() {
+                    pragma.foreign_keys(true);
```

Comment:
what if `pragma.foreign_keys(true);` returns non `SQLITE_OK` how user will get info about that?

Reply:
this is a problem with set_pragma which is called by all the setting pragmas; thus I suggest this solution

```C++
            template<class T>
            void set_pragma(const std::string& name, const T& value, sqlite3* db = nullptr) {
                auto con = this->get_connection();
                if(!db) {
                    db = con.get();
                }
                std::stringstream ss;
                ss << "PRAGMA " << name << " = " << value;
#if 1   // JDH
                T result;
                auto rc = sqlite3_exec(db, ss.str().c_str(), nullptr, &result, nullptr);
                if (rc != SQLITE_OK) {
                    throw_translated_sqlite_error(db);
                }
#else
                internal::perform_void_exec(db, ss.str());
#endif
            }

            void set_pragma(const std::string& name, const sqlite_orm::journal_mode& value, sqlite3* db = nullptr) {
                auto con = this->get_connection();
                if(!db) {
                    db = con.get();
                }
                std::stringstream ss;
                ss << "PRAGMA " << name << " = " << internal::to_string(value);
#if 1   // JDH
                auto rc = sqlite3_exec(db, ss.str().c_str(), nullptr, nullptr, nullptr);
                if (rc != SQLITE_OK) {
                    throw_translated_sqlite_error(db);
                }
#else
                internal::perform_void_exec(db, ss.str());
#endif
            }
        };
```

Do you agree?

Reply:
Actually perform_void_exec() checks return from sqlite3_exec as follows:

```C++
        inline void perform_void_exec(sqlite3* db, const std::string& query) {
            int rc = sqlite3_exec(db, query.c_str(), nullptr, nullptr, nullptr);
            if(rc != SQLITE_OK) {
                throw_translated_sqlite_error(db);
            }
        }
```

so nothing has to be changed!

Reply:
alright